src/main/java/usersandpostsCRUD/demo/service/PostService.java

```diff
@@ -0,0 +1,50 @@
+package usersandpostsCRUD.demo.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import usersandpostsCRUD.demo.dto.UserDto;
+import usersandpostsCRUD.demo.entity.Post;
+import usersandpostsCRUD.demo.entity.User;
+import usersandpostsCRUD.demo.exception.PostNotFoundException;
+import usersandpostsCRUD.demo.repository.PostRepository;
+import usersandpostsCRUD.demo.repository.UserRepository;
+
+import java.util.List;
+
+@Service
+public class PostService {
+
+    private final PostRepository postRepository;
+    private final UserRepository userRepository;
```

Comment:
UserRepository can only be injected in the UserService, nowhere else.
If you need to get the information about the user, you need to inject the UserService here.

Reply:
Not resolved.

User - entity class
UserRepository - repository class for the User entity
UserService - service class that can only autowire UserRepository.

You cannot autowire UserRepository in PostService.


You can only autowire UserServixe in the PostService.

If you need anything from the userRepository, theen you define what you need in the UserServixe and in PostServixe you AUTOWIRE UserService.